### DISCUSSION TITLE
Dimmer and Channel1 in DevGroup

### DISCUSSION THREAD

**Discussion by BusinessClaes** [arendst/Tasmota#16829] [Q&A] [answered]
Hi!

I have a devicegroup with several dimmer-devices and one channel1-device. I want them to share the dimmer value (as well as power and fade) so I set DevGroupShare 7,7 but since the channel1-device doesn’t respond to dimmer-command, it only follows the group when switching on/off.

Setting up a rule like “on dimmer#state do channel1 %value% endon” doesn’t solve the problem since the “device-shared-command” doesn’t seem to trigger any event in the channel1-device.

I wonder if there is some way of making channel1 follow dimmer that I am missing…

**Comment by pcdiem** [comment, marked as answer]
Can you explain what you mean by a "channel1-device"? I assume that since your device responds to the channel1 command, it must have a PWM GPIO defined. If it does have a PWM defined, it seems like it should respond to the dimmer command.

**Reply by BusinessClaes** [reply]
Hi! It’s a Shelly RGBW2 with only one PWM channel active
`{"NAME":"Shelly RGBW2","GPIO":[0,0,288,0,0,1,0,0,0,32,416,0,0,0],"FLAG":0,"BASE":18} `

As far as I can see it doesn’t respond to Dimmer command, only to Channel1 command to change the brightness of the attached led strip.

Do you mean that it should respond to Dimmer command, which means I’ve done something strange when configuring the device?

**Reply by barbudor** [reply]
What setoption did you used?
Specifically did you used setoption68 to used PWM channels independently? This may disable the dimmer command

**Reply by BusinessClaes** [reply]
You got it! Back now to So68 0 and everything working, the device now listens to Dimmer command.

I had activated so68 back in time, then I realized it was nicer to just wipe out in config the PWMs that I didn’t need. But I left so68 on… Thanks a lot for your help!

**Reply by barbudor** [reply]
Note that today you enabled only 1 pwm on your shelly
Once you will have more, tasmota behavior will change
By default with so68 off, all PWM channels belongs to the same Light.
2 channels and tasmota consider a cold white/warm white light
3 channels and that's a RGB light
4 channels got you a RGB+W
5 channels is for RGB+CW+WW
If that case, channel is still present and allow to control each PWM as well as Dimmer allow to control all n channels as a single light

With so68 1, the PWM channels are individualised and no more part of a single light. You would still have the channel command but no more the dimmer one
That's why mixing tasmota with different SO68 in the same dev group is not so simple

If you hadn't yet had the opportunity, read https://tasmota.github.io/docs/Lights/

**Comment by pcdiem** [comment]
I don't see why the channels received for a device group wouldn't update the corresponding channels even with SO68 1. When a device group channels item is received, it calls changeChannels which should call setChannelsRaw if SO68 is 1.

**Reply by barbudor** [reply]
Paul, you absolutely know much better than me how DevGroup behave 🤣 

Actually I didn't experienced it myself but on the above test from @BusinessClaes it seems that changing SO68 to Off did the trick.
Is that Channel or Dimmer that is transported through DevGroup ?
May be the problem is mixing lights that have SO68 Off and SO68 On ?
My explanation was just a guess ...

**Reply by BusinessClaes** [reply]
Just to clarify, the device sending the update to the DevGroup is a Shelly Dimmer2, so no PWM and I guess only Dimmer-info and no channel-info in the message sent. Maybe that’s the issue?

**Reply by pcdiem** [reply]
Device groups sends and receives channels as seen by the light module. There's really so difference at the channel level so an RGBCW light sending an update to a device group when the light color is changed to 100% blue should be received by an SO68 1 device as an update to set channel 3 to 255. I'll have to do some testing to find out why that's not happening.

**Reply by BusinessClaes** [reply]
Please tell me if you want me to (and how in that case) take out some debug info out of my setup. As i said, the device sending the update to the group is a Shelly dimmer2 and I think it doesn’t speak RGBW-PWM with its light module, I think its some communication protocol with the lighting co-processor. But I am now really out of my comfort zone, might be speaking nonsense… :D

**Reply by pcdiem** [reply]
Ah, right. The channels are sent and received scaled at 255 and the brightness/dimmer level is sent separately to indicate the brightness. So an RGBCW light sending an update when the light color is changed to 50% blue will send the channels as 0,0,255,0,0 and the brightness 128. The light module uses LightStateClass::setBri to set the brightness to the level received from a device group. I guess the logic there doesn't adjust the channels based on the brightness if SO68 is 1.

**Comment by BusinessClaes** [comment]
Going off topic now, but I have noticed some strange behaviour with my device group…

Background: I have a setup with a master (devgroupshare 0,7) and several slaves (devgroupshare 7,0) so I can control the whole group with the master or each slave device separately.

The issue: Having the whole group at a given dimmer value, then switching everything off with the master and then doing a dimmer 1 with the master, the slaves go first on to the previous dimmer value and then down to dimmer 1. Fast but very noticeable. The master sends a `MQT: stat/tasmota_857F7A/RESULT = {"POWER":"ON","Dimmer":1}`

The question: Is it possible to change the order of the message, first dimmer and then power?. Changing the whole tasmota core message feels a big thing, maybe it can be done at devgroup level?

**Reply by pcdiem** [reply]
I tried this here and see exactly what you're talking about. The reason is that the logic for the Dimmer command turns the power on first (causing a device group power on message to be sent) and then it set the new dimmer level.

**Reply by BusinessClaes** [reply]
Exactly, the question is if it is worth the pain and if there would be some unseen consequences of changing the order in the logic for the dimmer command…

In the meantime I come around the problem by doing a `backlog dimmer 1; power 0` in my rule for switch1#event instead of just `power 0`. So using my wall switch the problem doesn’t shown up although it’s still there.

**Reply by pcdiem** [reply]
I does seem like it would be better to change the power state after the brightness level is changed so that it turns on at the new level rather than turning on at the old level and then changing to the new level. I think this needs to be evaluated by the Light module developers to see if they agree.

**Reply by BusinessClaes** [reply]
Nice! What’s the way to go? Should I start a new thread at issues, or can we summon them here? :) (I don’t know how to do the latter though)

**Reply by barbudor** [reply]
Just let's call in @arendst for an advice : the question is about the order of Dimmer command and Power ON
It looks like Tasmota is powering first the light before setting the dimmer value, while the other way round could seems better.
What do you think Theo ?

**Reply by arendst** [reply]
In the old days this worked best. But it if it is a big issue try to swap em around and let's see what happens to the installed base.
